onnx/common/tensor.h

```diff
@@ -139,6 +139,221 @@ struct Tensor final {
     has_name_ = true;
     name_ = std::move(name);
   }
+
+  bool is_raw_data() const {
+    return is_raw_data_;
+  }
+
+  template<typename T, int64_t block_size>
+  void bin_func(void (*f)(void*, const void*), Tensor& a, std::vector<T>& T_data_, std::vector<T>& a_T_data_) {
+    int64_t num_elements = 1;
+    for (auto i : sizes_) {
+      num_elements *= i;
+    }
+    T* T_ptr;
+    const T* a_ptr;
+    if (is_raw_data_)  {
+      T_ptr = (T*) malloc(raw_data_.size());
```

Comment:
Explicit memory allocation here is not very safe. We can just we vector<T>.

Reply:
Agree

Reply:
changed to vector